denops/ddc/ddc.ts

```diff
@@ -126,58 +127,58 @@ export class Ddc {
     context: Context,
     options: DdcOptions,
   ): Promise<[number, DdcCandidate[]]> {
-    let completePos = -1;
-    let candidates: DdcCandidate[] = [];
-    const sources = options.sources.map((name) => this.sources[name])
-      .filter((x) => x);
-
-    for (const source of sources) {
-      const [sourceOptions, sourceParams] = sourceArgs(options, source);
-      completePos = await source.getCompletePosition(
+    const sources = this.foundSources(options.sources)
+      .map((s) => [s, ...sourceArgs(options, s)] as const);
+    const rs = await Promise.all(sources.map(async ([s, o, p]) => {
+      const completePos = await s.getCompletePosition(
         denops,
         context,
         options,
-        sourceOptions,
-        sourceParams,
+        o,
+        p,
       );
       const completeStr = context.input.slice(completePos);
       if (
-        completeStr.length < sourceOptions.minAutoCompleteLength ||
-        completeStr.length > sourceOptions.maxAutoCompleteLength
+        completeStr.length < o.minAutoCompleteLength ||
+        completeStr.length > o.maxAutoCompleteLength
       ) {
-        continue;
+        return;
```

Comment:
Why it uses return??
It seems wrong.

Reply:
No, it's correct. You cannot use `continue` on `map`.

Reply:
I get it.